### PR TITLE
fix: add package main field for TypeScript support

### LINKED ISSUE #979: Cannot find module 'fs-extra/esm' or its corresponding type declarations.ts(2307)
Both importing from fs-extra and fs-extra/esm doesn't work anymore in v11.0.0. I tried the following lines in my fs.ts file:

```
import { ensureDirSync, ensureFileSync, removeSync } from 'fs-extra'
import { ensureDirSync, ensureFileSync, removeSync } from 'fs-extra/esm'
```

The top line does work in v10.1.0. Am I doing something wrong?

- **Operating System:** Ubuntu 22.04.1 LTS
- **Node.js version:** v16.18.0
- **`fs-extra` version:** v11.0.0

### PR DESCRIPTION
**Pull request by mtdvlpr**
Fixes #979

### CODE REVIEW

**Review by caugner — APPROVED** [review]
See also: https://github.com/jprichardson/node-fs-extra/pull/980

Tested locally, and resolves the issue with TypeScript.

**Comment by caugner** [comment]
It doesn't resolve the issue with `import from "fs-extra/esm"` though. 🤔

**Comment by mtdvlpr** [comment]
It doesn't have to, right? Importing from fs-extra will get the esm version if I'm not mistaken.

**Comment by caugner** [comment]
> It doesn't have to, right? Importing from fs-extra will get the esm version if I'm not mistaken.

If I understand @RyanZim's intention correctly, the separate `fs-extra/esm` export aims to avoid confusion. Adding the "module" field reintroduces that confusion, because you could then import from `fs-extra` and get the ESM module.

We could just add the `main` field for now, and that alone would resolve TypeScript issue.

File: package.json [hunk 1]
```diff
@@ -51,6 +51,8 @@
     "read-dir-files": "^0.1.1",
     "standard": "^17.0.0"
   },
+  "main": "./lib/index.js",
+  "module": "./lib/esm.mjs",
```
Comment (caugner) [inline package.json:55]:
```suggestion
  "main": "./lib/index.js",
```

**Review by caugner — CHANGES_REQUESTED** [review]
A more conservative solution is to omit the `module` field for now, because the author does not intend to export the ESM module at `fs-extra`.

File: package.json (hunk 1 above)
Comment (mtdvlpr) [reply package.json:55]: Alright, but then it's impossible to get the esm module in TypeScript.

Comment (caugner) [reply package.json:55]: The solution for the ESM export (`import fse from fs-extra/esm`) might be to add a type declaration file, and replace the `./esm` entry in `exports` with an object that has two keys `import` and `types`.

Background on why the ESM export on `fs-extra` itself is undesired is here: https://github.com/jprichardson/node-fs-extra/pull/974#discussion_r1027022282

Comment (caugner) [reply package.json]: PS: I'm not actually sure what the advantage of the ESM module is (in this context). Afaik you can always import the CommonJS exports, even if you're in the context of a pure ESM package.

**Review by caugner — APPROVED** [review]
Re-tested and still resolves the issue when using `import fs from "fs-extra";` in TypeScript.

**Comment by RyanZim** [comment]
I removed the `main` field, since it wasn't needed for backward compatibility with old Node versions, as we don't support Node versions old enough not to support `exports`. However, since it seems TS doesn't like this (potentially a bug? see https://github.com/jprichardson/node-fs-extra/issues/979#issuecomment-1331404978), it seems wise to add it back for now.

> It doesn't resolve the issue with import ... from "fs-extra/esm" though. 

I think that's fine; `main` is only needed for TS support, and if you're using TS, you already have named import support; there's no reason to use `fs-extra/esm` in TS.

@caugner is correct; we don't want to add `module` here, as that would be confusing.

**Review by RyanZim — APPROVED** [review]

**Comment by RyanZim** [comment]
Published in v11.1.0 :tada:

**Pull request merged**